frontend/src/components/common/Typography.tsx

```diff
@@ -0,0 +1,49 @@
+import { PropsWithChildren, useState } from 'react'
+
+import { Typography as MaterialTypography, Theme, TypographyProps } from '@material-ui/core'
+import { createStyles, makeStyles } from '@material-ui/core/styles'
+import LinkIcon from '@material-ui/icons/Link'
+
+import LinkIconButton from 'components/common/LinkIconButton'
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    linkIcon: {
+      transform: 'rotate(-45deg)',
+      color: theme.palette.primary.main,
+      '&:hover': {
+        color: theme.palette.primary.dark,
+      },
+    },
+    hideLinkIcon: {
+      visibility: 'hidden',
+    },
+  }),
+)
+
+type TypographyParams = PropsWithChildren<TypographyProps> & {
+  id?: string
+  linkable?: boolean
+  component?: string
+}
+
+export default function Typography({ children, id, linkable, ...props }: TypographyParams) {
```

Comment:
I'd suggest that we rename the component from `Typography` to `<AnchorTypography />` or `<Heading />` as it currently overlaps with the generic component with the same name coming from MaterialUI